tree-building/tree_building.go

```diff
@@ -0,0 +1,128 @@
+package tree
+
+import "fmt"
+import "sort"
```

Comment:
These should be defined as

```go
import (
    "fmt"
    "sort"
)
```

You may want to set up [`goimports`](https://godoc.org/golang.org/x/tools/cmd/goimports) as it'll do this automatically :)